fix build containerd in centos9 (#7683)
containerd/containerd: BUILDING.md

`containerd` uses [Btrfs](https://en.wikipedia.org/wiki/Btrfs) it means that you
need to satisfy these dependencies in your system:

* CentOS 7 / Fedora: `yum install btrfs-progs-devel`
  * Note with CentOS 9:  BTRFS has been deprecated in RHEL / CentOS 7.4, and removed in RHEL/CentOS 9 .
    [Btrfs has been deprecated](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/storage_administration_guide/ch-btrfs).[release_notes](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/7.4_release_notes/chap-red_hat_enterprise_linux-7.4_release_notes-deprecated_functionality_in_rhel7#idm139789147351408)
* Debian/Ubuntu: `apt-get install btrfs-progs libbtrfs-dev`
  * Debian(before Buster)/Ubuntu(before 19.10): `apt-get install btrfs-tools`
* For unsupported [Btrfs](https://en.wikipedia.org/wiki/Btrfs) system:

>   * We suggest that using the no_btrfs build-flag to build without btrfs support.

Reviewer: ```suggestion
  * Use the `no_btrfs` build tag to build without btrfs support.
```

Author: @samuelkarp 
thanks for review.

>    

Reviewer: There's some trailing whitespace here, which is disallowed in the project checks:

> FAIL - has whitespace errors. See `git show --check 189384ca68730834ecbe09b59c56497a552070de`.

```suggestion
```

Author: thanks for review.
@samuelkarp

need to satisfy these dependencies in your system:

* CentOS/Fedora: `yum install btrfs-progs-devel`

>   * CentOS(CentOS Stream9): there have no btrfs-progs-devel anymore . You should rpm these packages manual for btrfs error on build.

Reviewer 2: So it is fine to use `btrfs-progs` as package name here?

Author: > Perhaps the better option is to describe that BTRFS has been deprecated in RHEL / CentOS 7.4, and removed in RHEL/CentOS 9 and up. (see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/storage_administration_guide/ch-btrfs, and the [release notes](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/7.4_release_notes/chap-red_hat_enterprise_linux-7.4_release_notes-deprecated_functionality_in_rhel7#idm139789147351408))
> 
> And for those platforms, recommend using the `no_btrfs` build-flag to build without btrfs support.

@thaJeztah @fuweid 
Thanks for review.
It sounds good to me..
I will update the building.md follow the infos..

    like : btrfs-progs-5.12.1-2.hs.el8.x86_64.rpm , btrfs-progs-devel-5.12.1-2.hs.el8.x86_64.rpm , libbtrfs-5.12.1-2.hs.el8.x86_64.rpm,libbtrfsutil-5.12.1-2.hs.el8.x86_64.rpm [CentOS Stream9](https://cbs.centos.org/koji/buildinfo?buildID=32944).

>     [build err](https://github.com/containerd/containerd/issues/3488).

Reviewer 2: I think we should delete this line and just add it in git message.
